src/ghostfat.c

```diff
@@ -270,11 +269,11 @@ void uf2_read_block (uint32_t block_no, uint8_t *data)
     for ( uint32_t i = 0; i < FAT_ENTRIES_PER_SECTOR; ++i )
     {
       // Generate the FAT chain for the firmware "file"
-      uint32_t v = (sectionIdx * FAT_ENTRIES_PER_SECTOR) + i;
+      uint32_t v = (sectionIdx * FAT_ENTRIES_PER_SECTOR * BPB_SECTORS_PER_CLUSTER) + (i * BPB_SECTORS_PER_CLUSTER);
 
       if ( UF2_FIRST_SECTOR <= v && v <= UF2_LAST_SECTOR )
       {
-        ((uint16_t*) (void*) data)[i] = v == UF2_LAST_SECTOR ? 0xffff : v + 1;
+        ((uint16_t*) (void*) data)[i] = v == UF2_LAST_SECTOR ? 0xffff : (v / BPB_SECTORS_PER_CLUSTER) + 1;
```

Comment:
Reverting this line is recommended.  Your changes were based on the very confusing `#define` names used in the prior code, coupled with your commit's change to the semantic meaning of two of the `#define`s above.

<details><summary>Comments to help refresh my memory</summary><P/>

Presuming the above change to use `UF2_FIRST_CLUSTER_NUMBER` and `UF2_LAST_CLUSTER_NUMBER`....

Here, `v` is the `fatArrayIndex` being modified by the particular loop iteration.  The array index inside the FAT uses the "cluster number" as its index.

The correct values to insert here are the `cluster number`.
This appears to be setting the FAT entry at index `v`


</details>

Revert this line, keeping the simpler:

```C
((uint16_t*) (void*) data)[i] = v == UF2_LAST_CLUSTER_NUMBER ? 0xffff : v + 1;
```